chutney/server/src/main/java/com/chutneytesting/execution/domain/campaign/CampaignExecutionEngine.java

```diff
@@ -304,7 +305,7 @@ private ExecutionRequest buildExecutionRequest(Campaign campaign, TestCaseDatase
             .or(() -> ofNullable(campaign.externalDatasetId))
             .map(datasetRepository::findById)
             .orElse(null);
-        return new ExecutionRequest(testCaseDataset.testcase(), campaign.executionEnvironment(), campaignExecution.userId, dataset, campaignExecution);
+        return new ExecutionRequest(testCaseDataset.testcase(), campaign.executionEnvironment(), campaignExecution.userId, dataset, campaignExecution, new HashSet<>(campaign.tags));
```

Comment:
Elsewhere tags are a List<String>, to be more consistent, i think it should be better to be a list too